ArjunaCore/arjuna/classes/com/arjuna/ats/arjuna/coordinator/BasicAction.java

```diff
@@ -2115,7 +2127,8 @@ protected int async_prepare(boolean reportHeuristics) {
 
         // Prepare 2PC aware resources
         while (pendingList.size() != 0) {
-            tasks.add(TwoPhaseCommitThreadPool.submitJob(new AsyncPrepare(this, reportHeuristics, pendingList.getFront())));
+            tasks.add(TwoPhaseCommitThreadPool.submitJob(this::explicitPhase2Prepare,
+                    new TwoPhaseCommitThreadPool.BAAsyncPrepareJobParams(this, pendingList.getFront(), reportHeuristics)));
```

Comment:
What was the motiviation to use an inner class to pass in the args in comparison to how you are doing it for the commit and abort? It would have ended up something like:
`tasks.add(TwoPhaseCommitThreadPool.submitJob(this::explicitPhase2Prepare, this, pendingList.getFront(), reportHeuristics)`
Which looks more like you do on 1562 and 1571